frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.tsx

```diff
@@ -1,76 +1,153 @@
+import TippyPopover from "metabase/components/Popover/TippyPopover";
 import Ellipsified from "metabase/core/components/Ellipsified";
 import Icon from "metabase/components/Icon";
 
 import type { DashboardOrderedCard } from "metabase-types/api";
+
+import { useToggle } from "metabase/hooks/use-toggle";
+import Search from "metabase/entities/search";
+
 import { isEmpty } from "metabase/lib/validate";
+import { color } from "metabase/lib/colors";
+
+import { EntityDisplay } from "./EntityDisplay";
 
-import { settings, LinkCardSettings } from "./LinkVizSettings";
+import { settings } from "./LinkVizSettings";
+
+import type { LinkEntity, LinkCardSettings } from "./types";
 
 import {
   EditLinkCardWrapper,
   DisplayLinkCardWrapper,
   CardLink,
+  SearchResultsContainer,
 } from "./LinkViz.styled";
 
+import { isUrlString } from "./utils";
+
 export interface LinkVizProps {
   dashcard: DashboardOrderedCard;
   isEditing: boolean;
-  isPreviewing: boolean;
-  isSettings: boolean;
   onUpdateVisualizationSettings: (
-    newSettings: Partial<LinkCardSettings>,
+    newSettings: Partial<DashboardOrderedCard["visualization_settings"]>,
   ) => void;
-  settings: LinkCardSettings;
+  settings: DashboardOrderedCard["visualization_settings"] & {
+    link: LinkCardSettings;
+  };
 }
 
 function LinkViz({
   dashcard,
   isEditing,
-  isPreviewing,
-  isSettings,
   onUpdateVisualizationSettings,
   settings,
 }: LinkVizProps) {
   const {
-    link: { url },
+    link: { url, entity },
   } = settings;
 
   const isNew = !!dashcard?.justAdded;
+  const [autoFocus, setAutoFocus] = useState(isNew);
+  const previousUrl = usePrevious(url);
+
+  const handleLinkChange = (newLink: string) =>
+    onUpdateVisualizationSettings({ link: { url: newLink } });
 
-  const handleLinkChange = (e: React.ChangeEvent<HTMLInputElement>) =>
-    onUpdateVisualizationSettings({ link: { url: e.currentTarget.value } });
+  const handleEntitySelect = (entity: LinkEntity) => {
+    onUpdateVisualizationSettings({
+      link: {
+        entity: {
+          id: entity.id,
+          db_id: entity.model === "table" ? entity.database_id : undefined,
+          name: entity.name,
+          model: entity.model,
+          description: entity.description,
+          display: entity.display,
+        },
+      },
+    });
+  };
 
-  const showEditBox = isEditing && !isPreviewing && !isSettings;
+  const [inputIsFocused, { turnOn: onFocusInput, turnOff: onBlurInput }] =
+    useToggle();
+
+  useEffect(() => {
+    // if the url was auto-filled from the entity, focus the input
+    if (previousUrl === undefined && !!url) {
+      setAutoFocus(true);
+    }
+  }, [previousUrl, url]);
+
+  if (entity) {
+    const wrappedEntity = Search.wrapEntity({
+      ...entity,
+      database_id: entity.db_id ?? entity.database_id,
+      table_id: entity.model === "table" ? entity.id : undefined,
+      collection: {},
+    });
```

Comment:
We need to wrap it, so we can use `getIcon` and `getUrl`, right?

Reply:
yes.